router/http/src/main/scala/io/buoyant/router/http/PathIdentifier.scala

```diff
@@ -0,0 +1,22 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle.{Dtab, Path, http}
+import com.twitter.finagle.buoyant.Dst
+import com.twitter.util.Future
+import io.buoyant.router.RoutingFactory
+
+case class PathIdentifier(
+  prefix: Path,
+  segments: Int = 1,
+  baseDtab: () => Dtab = () => Dtab.base
+) extends RoutingFactory.Identifier[http.Request] {
+
+  def apply(req: http.Request): Future[Dst] =
+    Path.read(req.path) match {
+      case path if path.size >= segments =>
+        Future.value(Dst.Path(prefix ++ path.take(segments), baseDtab(), Dtab.local))
+      case _ =>
+        Future.exception(new IllegalArgumentException(s"not enough segments in path ${req.path}"))
```

Comment:
You could make an argument for just taking as many segments as there are, instead of erroring out.  Not sure which is better, I could see it either way.